src/test/java/hudson/plugins/git/GitStatusTest.java

```diff
@@ -178,6 +184,47 @@ public void testDoNotifyCommitWithTwoBranchesAndAdditionalParameter() throws Exc
 
     }
 
+    @DataPoints("branchSpecPrefixes")
+    public static String[] branchSpecPrefixes = new String[] {
+            "",
+            "refs/remotes/",
```

Comment:
IMHO these a separate test cases because a regex or a wildcard defines the complete brancSpec and isn't just a prefix for a real branch name. These prefixes are used for the the tests: testDoNotifyBranchWithSlash, testDoNotifyBranchWithoutSlash, testDoNotifyBranchByBranchRef. I thinks it's not easy to find some regex or wildcard prefixes for the branches 'feature/awesome-feature' or 'awesome-feature' that are useful and will not break the tests.
For the regex cases I've already added the test 'testDoNotifyBranchWithRegex'. I'll add an additional one for a wildcard case, too.